packages/date-picker/src/DatePicker/DatePickerMenu/DatePickerMenuHeader/DatePickerMenuHeader.tsx

```diff
@@ -101,12 +91,7 @@ export const DatePickerMenuHeader = forwardRef<
       }
     };
 
-  /** Returns whether the provided month should be enabled */
-  const isMonthEnabled = useCallback(
-    (monthName: string) =>
-      shouldMonthBeEnabled(monthName, { month, min, max, locale }),
-    [locale, max, min, month],
-  );
+  const isYearFirstFormat = yearFirstFormats.includes(locale);
```

Comment:
I think checking the order of `formatParts` would be a more reliable way to solve this. It's _probably_ only ISO that formats year first, but I'm not 100% sure

Reply:
So basically, check if the year comes first in `formatParts`? There are a few other locales that format year first. This brings me back to the convo about `fa-AF` and how we're unsure if flipping the selects makes sense in that language.

Reply:
Since it's only iso that we went, I think i can just do a check `locale === 'iso8601'`